internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -1802,7 +1802,7 @@ func TestReconcileMachineDeploymentMachineHealthCheck(t *testing.T) {
 			current: nil,
 			desired: []*scope.MachineDeploymentState{
 				newFakeMachineDeploymentTopologyState("md-1", infrastructureMachineTemplate, bootstrapTemplate,
-					mhcBuilder.Build()),
+					mhcBuilder.Clone().Build()),
 			},
 			want: []*clusterv1.MachineHealthCheck{
 				mhcBuilder.
```

Comment:
Why do we only have the OwnerRef in this test? Is it because we're not cloning the builder here? (and that every subsequent cloned one already has it)